src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -187,6 +187,36 @@ public function paginate($perPage = null, $columns = array('*'))
 	}
 
 	/**
+	 * Chunk the results of the query.
+	 *
+	 * @param  int  $count
+	 * @param  callable  $callback
+	 * @return void
+	 */
+	public function chunk($count, callable $callback)
+	{
+		$this->query->addSelect($this->getSelectColumns());
+
+		$models = $this->query->forPage($page = 1, $count)->getModels();
+
+		while (count($models) > 0)
+		{
+			$this->hydratePivotRelation($models);
+
+			$result = $this->related->newCollection($models);
+
+			// On each chunk result set, we will pass them to the callback and then let the
+			// developer take care of everything within the callback, which allows us to
+			// keep the memory low for spinning through large result sets for working.
+			call_user_func($callback, $result);
```

Comment:
Shouldn't this check for `false` like [the query builder](https://github.com/laravel/framework/blob/5.0/src/Illuminate/Database/Query/Builder.php#L1482-L1485)?